release.config.mjs

```diff
@@ -0,0 +1,92 @@
+/**
+ * @type {import('semantic-release').GlobalConfig}
+ */
+export default {
+    branches: [
+        "main"
+    ],
+    plugins: [
+        [
+            "@semantic-release/commit-analyzer",
+            {
+                preset: "angular",
+                releaseRules: [
+                    { type: "feat", release: "minor" },
+                    { type: "refactor", scope: "major-change", release: "minor" },
+                    { type: "fix", release: "patch" },
+                    { type: "perf", release: "patch" },
+                    { type: "refactor", release: "patch" },
+                    { type: "style", release: "patch" },
+                    { type: "revert", release: "patch" },
+                    { type: "release", release: "major" },
+                    { breaking: true, release: "major" },
+                    { type: "build", release: false },
+                    { type: "chore", release: false },
+                    { type: "ci", release: false },
+                    { type: "docs", release: false },
+                    { scope: "no-release", release: false },
+                    { type: "test", release: false }
+                ]
+            }
+        ],
+        [
+            "@semantic-release/release-notes-generator",
+            {
+                preset: "conventionalcommits",
+                presetConfig: {
+                    types: [
+                        { type: "feat", section: "✨ Features", hidden: false },
```

Comment:
@stanleyrad1 Do we need all of these release types? I would think that the only ones that are important for this repo are "feat", and "fix", and maybe "refactor"

Reply:
I mean it doesn't hurt to have them, and let's you be more granular. If you notice in the commit-analyzer section, it will create a different type of release depending on how you prefix the commits. The release-notes-generator won't add sections that are not being used, if that's the concern

Reply:
Ah got it. I see in the releaseRules that a lot of these are turned off, that was my main concern. I didn't want a patch version to get created for a lot of these